assets/js/googlesitekit/data/index.js

```diff
@@ -53,5 +55,9 @@ Data.collectResolvers = collectResolvers;
 Data.collectSelectors = collectSelectors;
 Data.collectState = collectState;
 Data.collectName = collectName;
+Data.commonActions = commonActions;
+Data.commonControls = commonControls;
+Data.createRegistryControl = createRegistryControl;
+Data.createRegistrySelector = createRegistrySelector;
```

Comment:
How does this know it is supposed to use our `Data` as registry instead of `wp.data`? We're just importing and exporting that, and it looks to me like we're doing that without ensuring it uses our own registry. Or can you clarify how that works?

Reply:
The current registry provides itself to the registry control/selector during store registration.

These are documented, although it doesn't elaborate much on them (or how useful they are) in the docs:
* https://github.com/WordPress/gutenberg/tree/master/packages/data#createRegistryControl
* https://github.com/WordPress/gutenberg/tree/master/packages/data#createRegistrySelector

Basically, each takes a function that receives the current registry, and then returns a normal control/selector function.

It's more clear to see how this works for controls, but it's basically the same for registry selectors:
https://github.com/WordPress/gutenberg/blob/4857ad58c1241b3d63d21a6880c989b85746c3dc/packages/data/src/namespace-store/index.js#L129-L134

The main takeaway is that the `create*` function doesn't have access to the `registry` at the time it's called. It only "marks" the given function as a registry control/selector so that it is handled differently during registration when the current `registry` is available and then invokes the given callback with the registry instance to get back the "normal" function.